tests/Doctrine/Tests/ORM/Functional/PaginationTest.php

```diff
@@ -58,33 +57,144 @@ public function testCountComplexWithOutputWalker()
 
-        $paginator = new Paginator($query, true);
+        $paginator = new Paginator($query, $fetchJoinCollection);
+        $paginator->setUseOutputWalkers($useOutputWalkers);
+        $this->assertCount(9, $paginator->getIterator());
+
+        // Test with limit
+        $query->setMaxResults(3);
+        $paginator = new Paginator($query, $fetchJoinCollection);
         $paginator->setUseOutputWalkers($useOutputWalkers);
         $this->assertCount(3, $paginator->getIterator());
+
+        // Test with limit and offset
+        $query->setMaxResults(3)->setFirstResult(4);
+        $paginator = new Paginator($query, $fetchJoinCollection);
+        $paginator->setUseOutputWalkers($useOutputWalkers);
+        $this->assertCount(3, $paginator->getIterator());
+    }
+
+    private function iterateWithOrder($useOutputWalkers, $fetchJoinCollection, $baseDql, $checkField)
+    {
+        // Ascending
+        $dql = "$baseDql ASC";
+        $query = $this->_em->createQuery($dql);
+
+        $paginator = new Paginator($query, $fetchJoinCollection);
+        $paginator->setUseOutputWalkers($useOutputWalkers);
+        $iter = $paginator->getIterator();
+        $this->assertCount(9, $iter);
+        $result = iterator_to_array($iter);
+        $this->assertEquals($checkField . "0", $result[0]->$checkField);
+
+        // With limit
+        $query->setMaxResults(3);
+        $paginator = new Paginator($query, $fetchJoinCollection);
+        $paginator->setUseOutputWalkers($useOutputWalkers);
+        $iter = $paginator->getIterator();
+        $this->assertCount(3, $iter);
+        $result = iterator_to_array($iter);
+        $this->assertEquals($checkField . "0", $result[0]->$checkField);
+
+        // With offset
+        $query->setMaxResults(3)->setFirstResult(3);
+        $paginator = new Paginator($query, $fetchJoinCollection);
+        $paginator->setUseOutputWalkers($useOutputWalkers);
+        $iter = $paginator->getIterator();
+        $this->assertCount(3, $iter);
+        $result = iterator_to_array($iter);
+        $this->assertEquals($checkField . "3", $result[0]->$checkField);
+
+        // Descending
+        $dql = "$baseDql DESC";
+        $query = $this->_em->createQuery($dql);
+
+        $paginator = new Paginator($query, $fetchJoinCollection);
+        $paginator->setUseOutputWalkers($useOutputWalkers);
+        $iter = $paginator->getIterator();
+        $this->assertCount(9, $iter);
+        $result = iterator_to_array($iter);
+        $this->assertEquals($checkField . "8", $result[0]->$checkField);
+
+        // With limit
+        $query->setMaxResults(3);
+        $paginator = new Paginator($query, $fetchJoinCollection);
+        $paginator->setUseOutputWalkers($useOutputWalkers);
+        $iter = $paginator->getIterator();
+        $this->assertCount(3, $iter);
+        $result = iterator_to_array($iter);
+        $this->assertEquals($checkField . "8", $result[0]->$checkField);
```

Comment:
You are making a bunch of unrelated tests in this method. These should be separate tests instead (otherwise a failure for one of them will prevent all other tests from running, making it harder to identify what works and what breaks)